src/Serializer.php

```diff
@@ -118,8 +133,172 @@ public static function decodeMetadata($metadata)
         return $result;
     }
 
+    private function encodeElement(FieldDescriptor $field, $data)
+    {
+        switch ($field->getType()) {
+            case GPBType::MESSAGE:
+                if (is_array($data)) {
+                    $result = $data;
+                } else {
+                    if (!($data instanceof Message)) {
+                        throw new RuntimeException("Expected message, instead got " . get_class($data));
+                    }
+                    $result = $this->encodeMessageImpl($data, $field->getMessageType());
+                }
+                $messageType = $field->getMessageType()->getFullName();
+                if (isset($this->messageTypeTransformers[$messageType])) {
+                    $result = $this->messageTypeTransformers[$messageType]($result);
+                }
+                break;
+            default:
+                $result = $data;
+                break;
+        }
+
+        if (isset($this->fieldTransformers[$field->getName()])) {
+            $result = $this->fieldTransformers[$field->getName()]($result);
+        }
+        return $result;
+    }
+
+    private function encodeMessageImpl(Message $message, Descriptor $messageType)
+    {
+        $data = [];
+
+        foreach ($messageType->getField() as $field) {
+            /** @var FieldDescriptor $field */
```

Comment:
If it's useful for something, then feel free to keep it - I figured that was the case, but I'm not familiar enough with PHP development to know :-)